internal/controller/compaction/reconciler.go

```diff
@@ -493,3 +496,16 @@ func getCompactionJobArgs(etcd *druidv1alpha1.Etcd, metricsScrapeWaitDuration st
 
 	return command
 }
+
+func getEtcdCompactionAnnotations(etcdAnnotations map[string]string) map[string]string {
+	etcdCompactionAnnotations := make(map[string]string)
+
+	for key, value := range etcdAnnotations {
+		// Do not add annotation: `cluster-autoscaler.kubernetes.io/safe-to-evict: "false"` to compaction job's pod template.
+		if key == SafeToEvictKey && value == "false" {
```

Comment:
Right now, I want functionality to be correct, so I'm specifically checking and ignoring this annotation only `cluster-autoscaler.kubernetes.io/safe-to-evict: "false"` only.

>Can we just check only key because we don't set SafeToEvictKey:"true" to etcd anyway.

we don't know, may be somebody want to set this in future, as it's kind of depend on consumer of druid as in our case it's g/g which creates Etcd CR and annotation is used by CA.
Right now, I want to be functionality to be exact to what we want.